packages/web-app-files/src/services/folder/loaderSharedViaLink.ts

```diff
@@ -40,13 +42,23 @@ export class FolderLoaderSharedViaLink implements FolderLoader {
       })
 
       resources = resources.map((r) => r.shareInfo)
+      let spaces = []
+      if (store.getters.capabilities?.spaces?.enabled) {
+        const accessToken = store.getters['runtime/auth/accessToken']
+        const serverUrl = configurationManager.serverUrl
+        const graphClient = clientService.graphAuthenticated(serverUrl, accessToken)
+        // FIXME: Wait until spaces are loaded? We already load them in the runtime
+        yield store.dispatch('runtime/spaces/loadSpaces', { graphClient })
```

Comment:
Yes, please wait until spaces are loaded. Don't load them again. You can achieve that by adding the `spacesLoading` getter from `runtime/spaces` to the app loading spinner in the view template.

Reply:
discussed it, will be fixed in https://github.com/owncloud/web/pull/7430 (as that has more tooling in place regarding spaces in the store).